storm/src/main/storm/mesos/util/MesosCommon.java

```diff
@@ -105,7 +105,7 @@ public static String taskId(String nodeid, int port) {
   }
 
   public static String supervisorId(String nodeid, String topologyId) {
-    return String.format("%s-%s", nodeid, topologyId);
+    return String.format("%s%s%s", nodeid, DEFAULT_MESOS_COMPONENT_NAME_DELIMITER, topologyId);
```

Comment:
Name and ID are different things.  The configurability only relates to the name, which is a human readable nice string.  We shouldn't have any logic dependent on the format of the name itself.